src/HDF5.jl

```diff
@@ -1370,12 +1378,50 @@ Base.getindex(parent::Union{File,Group}, r::Reference) = _deref(parent, r)
 Base.getindex(parent::Dataset, r::Reference) = _deref(parent, r) # defined separately to resolve ambiguity
 
 # convert special types to native julia types
-normalize_types(x) = x
-normalize_types(x::NamedTuple{T}) where {T} = NamedTuple{T}(map(normalize_types, values(x)))
-normalize_types(x::Cstring) = unsafe_string(x)
-normalize_types(x::FixedString) = unpad(String(collect(x.data)), pad(x))
-normalize_types(x::FixedArray) = normalize_types.(reshape(collect(x.data), size(x)...))
-normalize_types(x::VariableArray) = normalize_types.(copy(unsafe_wrap(Array, convert(Ptr{eltype(x)}, x.p), x.len, own=false)))
+function normalize_types(::Type{T}, buf::AbstractMatrix{UInt8}) where {T}
+    return [_normalize_types(T, view(buf, :, ind)) for ind in axes(buf, 2)]
+end
+
+# high-level description which should always work
+function _typed_load(::Type{T}, buf::AbstractVector{UInt8}) where {T}
+    return @inbounds reinterpret(T, buf)[1]
```

Comment:
To confirm this will always be the first element, since `buf` will exactly always be  8 elements long and  T will always be an 8-byte pointer.  (at least on x64)? Maybe then just use `[]` ? 

Also, could we add comments above on what _typed_load is doing i.e. "reinterpret the vector buffer as a pointer locating the data. Note, by construction the buffer will exactly match the pointer size."

Reply:
Yes, I can add a comment because it's not a pointer that is being looked for — "the buffer contains the bytes for exactly 1 element of an object of type `T`, so reinterpret `UInt8` vector as a length-1 array of type `T`."